turbot/steampipe-plugin-azure
# Update Azure SDK version

- Update SDK to v68(latest)
- Add authentication method using `azidentity` and `resourcemanager` (GetNewSessionUpdated function)
- Add cache to the GetNewSessionUpdated function
- Keep the existing SPC requirement
- Keep the existing GetNewSession function
- Update the tables which are failing due to SDK update to use the GetNewSessionUpdated
- Manually check the failed tables after the pkg update and list the ones with breaking changes
   - azure_kubernetes_service_version (Breaking change but not used by any Azure Mod)
   Current columns - 
```
   > .inspect azure_kubernetes_service_version
+----------------------+---------+-------------------------------------------------------------------------------+
| column               | type    | description                                                                   |
+----------------------+---------+-------------------------------------------------------------------------------+
| _ctx                 | jsonb   | Steampipe context in JSON form, e.g. connection_name.                         |
| akas                 | jsonb   | Array of globally unique identifier strings (also known as) for the resource. |
| cloud_environment    | text    | The Azure Cloud Environment.                                                  |
| default              | boolean | Installed by default if version is not specified.                             |
| id                   | text    | ID of the orchestrator version profile list result.                           |
| is_preview           | boolean | Whether Kubernetes version is currently in preview.                           |
| location             | text    | The Azure region/location in which the resource is located.                   |
| name                 | text    | Name of the orchestrator version profile list result.                         |
| orchestrator_type    | text    | The orchestrator type.                                                        |
| orchestrator_version | text    | Orchestrator version (major, minor, patch).                                   |
| resource_type        | text    | Whether Kubernetes version is currently in preview.                           |
| subscription_id      | text    | The Azure Subscription ID in which the resource is located.                   |
| title                | text    | Title of the resource.                                                        |
| type                 | text    | Type of the orchestrator version profile list result.                         |
| upgrades             | jsonb   | The list of available upgrade versions.                                       |
+----------------------+---------+-------------------------------------------------------------------------------+
```
   Updated Column Set -
```
> .inspect azure_kubernetes_service_version
+-------------------+---------+-------------------------------------------------------------+
| column            | type    | description                                                 |
+-------------------+---------+-------------------------------------------------------------+
| _ctx              | jsonb   | Steampipe context in JSON form, e.g. connection_name.       |
| capabilities      | jsonb   | Capabilities on this Kubernetes version.                    |
| cloud_environment | text    | The Azure Cloud Environment.                                |
| is_preview        | boolean | Whether Kubernetes version is currently in preview.         |
| location          | text    | The Azure region/location in which the resource is located. |
| patch_versions    | jsonb   | Patch versions of Kubernetes release.                       |
| subscription_id   | text    | The Azure Subscription ID in which the resource is located. |
| title             | text    | Title of the resource.                                      |
| version           | text    | The major.minor version of Kubernetes release.              |
+-------------------+---------+-------------------------------------------------------------+
```   
- Update the deprecated pkgs for all the tables
- Run integration test. - Pending
